src/style-spec/expression/evaluation_context.js

```diff
@@ -34,27 +34,27 @@ class EvaluationContext {
         this.featureDistanceData = null;
     }
 
-    id() {
+    id(): null | any {
         return this.feature && 'id' in this.feature ? this.feature.id : null;
     }
 
-    geometryType() {
+    geometryType(): null | string {
         return this.feature ? typeof this.feature.type === 'number' ? geometryTypes[this.feature.type] : this.feature.type : null;
     }
 
     geometry() {
         return this.feature && 'geometry' in this.feature ? this.feature.geometry : null;
     }
 
-    canonicalID() {
+    canonicalID(): null | CanonicalTileID {
         return this.canonical;
     }
 
-    properties() {
+    properties(): {[string]: any} {
```

Comment:
If we don't rely on the property value types anywhere, can we try `mixed` instead of `any` here?

Reply:
Unfortunately, we rely on the properties in the `VectorTileFeature` interface and in the style-spec expression `Value`

https://github.com/mapbox/mapbox-gl-js/blob/bef7956206b070a762eb59c40f52613eac73cf88/flow-typed/vector-tile.js#L22

https://github.com/mapbox/mapbox-gl-js/blob/bef7956206b070a762eb59c40f52613eac73cf88/src/style-spec/expression/values.js#L32